src/java.base/share/classes/java/text/CollationElementIterator.java

```diff
@@ -76,17 +76,15 @@
  * <p>
  * Example of the iterator usage,
  * <blockquote>
- * <pre>
- *
- *  String testString = "This is a test";
- *  Collator col = Collator.getInstance();
- *  if (col instanceof RuleBasedCollator) {
- *      RuleBasedCollator ruleBasedCollator = (RuleBasedCollator)col;
- *      CollationElementIterator collationElementIterator = ruleBasedCollator.getCollationElementIterator(testString);
- *      int primaryOrder = CollationElementIterator.primaryOrder(collationElementIterator.next());
- *          :
- *  }
- * </pre>
+ * {@snippet lang=java :
+ * String testString = "This is a test";
+ * Collator col = Collator.getInstance();
+ * if (col instanceof RuleBasedCollator) {
```

Comment:
I'd use the pattern matching `instanceof` here

Reply:
Replaced the casting conversion with the pattern matching as you suggested, thank you